pkg/server/biz/accelerator/cleaner/clean.go

```diff
@@ -0,0 +1,249 @@
+package cleaner
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/caicloud/nirvana/log"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/caicloud/cyclone/pkg/k8s/clientset"
+	"github.com/caicloud/cyclone/pkg/meta"
+	"github.com/caicloud/cyclone/pkg/server/apis/v1alpha1"
+	"github.com/caicloud/cyclone/pkg/server/common"
+	"github.com/caicloud/cyclone/pkg/server/config"
+)
+
+// Cleaner cleans acceleration caches of a project
+type Cleaner struct {
+	clusterClient    kubernetes.Interface
+	client           clientset.Interface
+	projectName      string
+	projectNamespace string
+	config           *config.CacheCleaner
+}
+
+// NewCleaner creates a new acceleration caches cleaner.
+func NewCleaner(clusterClient kubernetes.Interface, client clientset.Interface, projectNamespace, projectName string) *Cleaner {
+	return &Cleaner{
+		clusterClient:    clusterClient,
+		client:           client,
+		projectName:      projectName,
+		projectNamespace: projectNamespace,
+		config:           &config.Config.CacheCleaner,
+	}
+}
+
+// Clean start a pod to do the clean work, and returns the pod name.
+func (c *Cleaner) Clean(pvcNamespace, pvcName string) (*v1alpha1.AccelerationCacheCleanupStatus, error) {
+	// Create a cache clean pod to clean acceleration cache data on PV.
+	cacheCleanPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.podName(c.projectName),
+			Namespace: pvcNamespace,
+			Labels: map[string]string{
+				// meta.LabelWorkflowRunName: o.wfr.Name,
+				meta.LabelPodKind:      meta.PodKindAccelerationGC.String(),
+				meta.LabelPodCreatedBy: meta.CycloneCreator,
+			},
+			Annotations: map[string]string{
+				meta.AnnotationIstioInject: meta.AnnotationValueFalse,
+			},
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers: []corev1.Container{
+				{
+					Name:    common.CacheCleanupContainerName,
+					Image:   c.config.Image,
+					Command: []string{"rm", "-rf", fmt.Sprintf("/%s/%s", common.CachePrefixPath, c.projectName)},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      common.DefaultCacheVolumeName,
+							MountPath: fmt.Sprintf("/%s", common.CachePrefixPath),
+							SubPath:   common.CachePrefixPath,
+						},
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse(getOrDefault(c.config, corev1.ResourceRequestsCPU, "50m")),
```

Comment:
Define the default value along with config definition in `pkg/server/config/config.go` instead of here.